include/cartesian_geom/autopoint.h

```diff
@@ -0,0 +1,311 @@
+    }
+
+    autopoint pow(int n) const {
+        autopoint temp;
+        temp.d = d;
+        temp.coeffs=coeffs.array().pow(n).matrix();
+        return temp;
+    }
+
+
+    autopoint log() const {
+        autopoint temp;
+        temp.d = d;
+        temp.coeffs=coeffs.array().log().matrix();
+        return temp;
+    }
+    
+    autopoint exp() const {
+        autopoint temp;
+        temp.d = d;
+        temp.coeffs=coeffs.array().exp().matrix();
+        return temp;
+    }
+    void operator+= (const autopoint& p)
+    {
+        coeffs += p.getCoefficients();
+    }
+
+    void operator+= (const Coeff& coeffs)
+    {
+        this->coeffs += coeffs ;
+    }
+
+    void operator-= (const autopoint& p)
+    {
+        coeffs -= p.getCoefficients();
+    }
+
+    void operator-= (const Coeff& coeffs)
+    {
+        this->coeffs -= coeffs ;
+    }
+
+
+    void operator= (const Coeff& coeffs)
+    {
+        this->coeffs = coeffs;
+        d = coeffs.rows();
+    }
+
+    // copy assignment
+autopoint& operator=(const autopoint& coeffs)
+{
+    // Guard self assignment
+    if (this == &coeffs)
+        return *this;
+ 
+    
+    this->coeffs = coeffs;
+    d = coeffs.rows();
+    return *this;
+}
+    autopoint transpose() const
+    {
+        return autopoint((Coeff)coeffs.transpose());
+    }
+
+    autopoint operator+ (const autopoint& p) const
+    {
+        return autopoint((Coeff)(coeffs+p.getCoefficients()));
+    }
+        autopoint operator- (const autopoint& p) const
+    {
+        return autopoint((Coeff)(coeffs-p.getCoefficients()));
+    }
+
+    autopoint operator- (const FT p) const
+    {
+         autopoint temp_auto;
+         temp_auto.d=d;
+         auto temp=coeffs.array();
+         temp_auto.coeffs=(temp-p).matrix();
+         return temp_auto;
+    }
+    autopoint operator- (T p) const
+    {
+         autopoint temp_auto;
+         temp_auto.d=d;
+         auto temp=coeffs.array();
+         temp_auto.coeffs=(temp-p).matrix();
+         return temp_auto;
+    }
+
+    autopoint operator* (const FT k)
+    {
+        return autopoint(coeffs * k);
+    }
+
+    autopoint operator* (T k) const
+    {
+        return autopoint((Coeff)(coeffs * ((FT)k)));
+    }
+    
+    autopoint operator* (T k)
+    {
+        return autopoint((Coeff)(coeffs * ((FT)k)));
+    }
+    // matrix multiplication
+    autopoint operator* (const autopoint& autopoint_)
+    {
+        return autopoint((Coeff)(coeffs * autopoint_.getCoefficients())); 
+    }
+    // matrix multiplication with normal matrix
+    autopoint operator* (const coeff& matrix_)
+    {
+        return autopoint((  autopoint(matrix_) * autopoint(coeffs) )); 
+    }
+
+    void operator*= (const FT k)
+    {
+        coeffs *= k;
+    }
+
+    void operator*= (const T k)
+    {
+        FT k_=k;
+        coeffs *= k_;
+    }
+
+
+    void operator/= (const FT k)
+    {
+        coeffs /= k;
+    }
+
+
+    FT distance(autopoint const & p) {
```

Comment:
you use a mixing style allover the file here (also in other files). E.g. is `{` in the same line as the function definition? is it `const autopoint&` or `autopoint const &` ? etc 
If in doubt see the style in other similar files of the library

Reply:
Thanks for these pointers, I have never considered these before and I have rectified all the { .